src/java/org/apache/cassandra/index/sai/QueryContext.java

```diff
@@ -265,8 +269,9 @@ public ScoreStoreProxy getScoreStoreProxyForSSTable(SSTableId<?> sstableId)
     public void recordScore(PrimaryKey primaryKey, float score)
     {
         // The stored score should never be less than 0. If it is, we have to recompute similarity later, which is why
-        // this is only an assert and not an illegal argument.
-        assert score >= 0;
+        // this is only an assert and not an illegal argument. Note that we cannot drop this score due to shadow keys.
+        if (score < 0.0f)
```

Comment:
Do we want to put more information here so that if the message is encountered someone can figure out why it happened?  Not sure this one is actual useful if it is seen